src/cert/client_cert_store.cpp

```diff
@@ -34,6 +35,32 @@ namespace
 {
 constexpr auto chain_name = "multipass_client_certs.pem";
 
+auto load_certs_from_file(const multipass::Path& cert_dir)
+{
+    std::vector<std::string> certs;
+    std::string cert;
+    auto path = QDir(cert_dir).filePath(chain_name);
+
+    if (QFile::exists(path))
+    {
+        std::ifstream cert_file{path.toStdString()};
+        std::string line;
+        while (std::getline(cert_file, line))
+        {
+            // Re-add the newline to make comparing of cert sent by client easier
+            cert.append(line + '\n');
+
+            if (line == "-----END CERTIFICATE-----")
+            {
+                certs.push_back(cert);
+                cert.clear();
+            }
+        }
```

Comment:
How will this deal with potentially corrupted files?

Reply:
Right, still need to deal with some error handling.  But this certainly isn't the only place in the Multipass code that doesn't take corrupted files into consideration :wink:

Reply:
Ok, added some error handling.